test/local/cache.js

```diff
@@ -0,0 +1,276 @@
+    hash.update(token.uid)
+    hash.update(token.id)
+    digest = hash.digest('base64')
+  })
+
+  afterEach(() => sandbox.restore())
+
+  it('exports the correct interface', () => {
+    assert.ok(cache)
+    assert.equal(typeof cache, 'object')
+    assert.equal(Object.keys(cache).length, 3)
+    assert.equal(typeof cache.del, 'function')
+    assert.equal(cache.del.length, 1)
+    assert.equal(typeof cache.get, 'function')
+    assert.equal(cache.get.length, 1)
+    assert.equal(typeof cache.set, 'function')
+    assert.equal(cache.set.length, 2)
+  })
+
+  describe('memcached resolves:', () => {
+    beforeEach(() => {
+      sandbox.stub(Memcached.prototype, 'delAsync', () => P.resolve())
+      sandbox.stub(Memcached.prototype, 'getAsync', () => P.resolve('mock get result'))
+      sandbox.stub(Memcached.prototype, 'setAsync', () => P.resolve())
+    })
+
+    describe('del:', () => {
+      beforeEach(() => {
+        return cache.del(token)
+      })
+
+      it('calls memcached.delAsync correctly', () => {
+        assert.equal(Memcached.prototype.delAsync.callCount, 1)
+        const args = Memcached.prototype.delAsync.args[0]
+        assert.equal(args.length, 1)
+        assert.equal(args[0], digest)
+
+        assert.equal(Memcached.prototype.getAsync.callCount, 0)
+        assert.equal(Memcached.prototype.setAsync.callCount, 0)
+        assert.equal(log.error.callCount, 0)
+      })
+    })
+
+    describe('del with bad token:', () => {
+      let error
+
+      beforeEach(() => {
+        return cache.del({ id: token.id })
+          .catch(e => error = e)
+      })
+
+      it('rejects correctly', () => {
+        assert.equal(error.message, 'Invalid token')
+      })
+
+      it('does not call memcached.delAsync', () => {
+        assert.equal(Memcached.prototype.delAsync.callCount, 0)
+      })
+    })
+
+    describe('get:', () => {
+      let result
+
+      beforeEach(() => {
+        return cache.get(token)
+          .then(r => result = r)
+      })
+
+      it('returns the correct result', () => {
+        assert.equal(result, 'mock get result')
+      })
+
+      it('calls memcached.getAsync correctly', () => {
+        assert.equal(Memcached.prototype.getAsync.callCount, 1)
+        const args = Memcached.prototype.getAsync.args[0]
+        assert.equal(args.length, 1)
+        assert.equal(args[0], digest)
+
+        assert.equal(Memcached.prototype.delAsync.callCount, 0)
+        assert.equal(Memcached.prototype.setAsync.callCount, 0)
+        assert.equal(log.error.callCount, 0)
+      })
+    })
+
+    describe('get with bad token:', () => {
+      let error
+
+      beforeEach(() => {
+        return cache.get({ uid: token.uid })
+          .catch(e => error = e)
+      })
+
+      it('rejects correctly', () => {
+        assert.equal(error.message, 'Invalid token')
+      })
+
+      it('does not call memcached.getAsync', () => {
+        assert.equal(Memcached.prototype.getAsync.callCount, 0)
```

Comment:
Are you splitting up the asserts in certain spots for readability? Seems like some could be combined and pass the message to the assert to give it more context.